verilog/tools/kythe/vname_context.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_TOOLS_KYTHE_VNAME_CONTEXT_H_
+#define VERIBLE_VERILOG_TOOLS_KYTHE_VNAME_CONTEXT_H_
+
+#include "common/util/auto_pop_stack.h"
+#include "verilog/tools/kythe/kythe_facts.h"
+
+namespace verilog {
+namespace kythe {
+
+// Container with a stack of VNames to hold context of VNames during traversal
+// of an IndexingFactsTree.
```

Comment:
If this is only used by one function/class, you could define this as a private class member, e.g.:

```c++
// header
class Foo {

private:
  class VNameContext;  // forward declaration
};
```

```c++
// implementation
class Foo::VNameContext {
  // full class definition
};
```